hr_timesheet_holiday/tests/test_all.py

```diff
@@ -0,0 +1,192 @@
+        account = project.analytic_account_id
+        account.write({"is_leave_account": True})
+        # Link sick leave to analytic account
+        sl = self.sl
+        sl.write({"analytic_account_id": account.id})
+
+        hol_empl_grp = self.leave.with_user(self.user_employee_id)
+        leave = hol_empl_grp.create(
+            {
+                "name": "One week sick leave",
+                "employee_id": self.employee_emp_id,
+                "holiday_status_id": self.sl.id,
+                "date_from": time.strftime("1900-01-06"),
+                "date_to": time.strftime("1900-01-12"),
+            }
+        )
+        with self.assertRaises(UserError):
+            leave.action_approve()
+
+    def test_allocation(self):
+        # Create analytic account
+        project = self.project.create(
+            {
+                "name": "Allocation",
+                "allow_timesheets": True,
+            }
+        )
+        project._create_analytic_account()
+        account = project.analytic_account_id
+        account.write({"is_leave_account": True})
+        # Link sick leave to analytic account
+        sl = self.sl
+        sl.write({"analytic_account_id": account.id})
+        leave = self.leave.create(
+            {
+                "name": "One week sick leave",
+                "holiday_status_id": self.sl.id,
+                "date_from": time.strftime("%Y-%m-06"),
+                "date_to": time.strftime("%Y-%m-12"),
+                "employee_id": self.employee.id,
+            }
+        )
+        leave.action_approve()
+        self.assertEqual(
+            len(leave.analytic_line_ids),
+            4,
+            "Allocation should not have " "analytic lines",
+        )
+
+    def test_timesheet_half_day(self):
+        # Test partial day leaves creates timesheet entries
+        self.env.ref("base.main_company").timesheet_hours_per_day = 7.0
+        project = self.project.create(
+            {
+                "name": "Test Project 1",
+                "allow_timesheets": False,
+            }
+        )
+        project._create_analytic_account()
+        account = project.analytic_account_id
+        project.write({"allow_timesheets": True})
+        account.write({"is_leave_account": True})
+        # Confirm leave and check hours added to account
+        hours_before = sum(account.line_ids.mapped("amount"))
+        # Holidays.with_user(self.user_employee_id)
+        hol_empl_grp = self.leave.with_user(self.user_hruser_id)
+        leave_date = datetime.today() + (relativedelta(days=10))
+        leave = hol_empl_grp.create(
+            {
+                "name": "One day and a half sick leave",
+                "employee_id": self.employee_emp_id,
+                "holiday_status_id": self.sl.id,
+                "date_from": leave_date - relativedelta(hours=10.5),
+                "date_to": leave_date,
+            }
+        )
+        leave.with_user(self.user_hruser_id).action_approve()
```

Comment:
this is the line that is failing:

`2024-02-28 05:30:33,674 399 ERROR odoo odoo.addons.hr_timesheet_holiday.tests.test_all: ERROR: TimesheetHolidayTest.test_timesheet_half_day
Traceback (most recent call last):
  File "/__w/timesheet/timesheet/hr_timesheet_holiday/tests/test_all.py", line 190, in test_timesheet_half_day
    leave.with_user(self.user_hruser_id).action_approve()
  File "/__w/timesheet/timesheet/hr_timesheet_holiday/models/hr_leave.py", line 64, in action_approve
    res = super(HrLeave, self).action_approve()
  File "/opt/odoo/addons/hr_holidays/models/hr_leave.py", line 1133, in action_approve
    self.filtered(lambda hol: not hol.validation_type == 'both').action_validate()
  File "/opt/odoo/addons/hr_holidays/models/hr_leave.py", line 1157, in action_validate
    raise ValidationError(_('The following employees are not supposed to work during that period:\n %s') % ','.join(leaves.mapped('employee_id.name')))
odoo.exceptions.ValidationError: The following employees are not supposed to work during that period:
 David Employee
 `

I think that it is because it is creating a leave on  a weekend, but not sure